usaspending_api/references/v2/views/cfda.py

```diff
@@ -57,7 +64,13 @@ def _populate_cfdas_if_needed(self):
             CFDA_DICTIONARY = response
 
     def _request_from_grants_api(self):
-        return post(
-            "https://www.grants.gov/grantsws/rest/opportunities/search/cfda/totals",
+        cfda_response = post(
+            CFDA_URL,
             headers={"Authorization": f"APIKEY={settings.GRANTS_API_KEY}"},
-        ).json()["cfdas"]
+        )
+        if cfda_response.status_code == status.HTTP_503_SERVICE_UNAVAILABLE:
+            raise ServiceUnavailable(f"{CFDA_URL} not available (status 503)")
+        if cfda_response.json()["errorMsgs"] != []:
+            raise InternalServerError(f"Error returned by {CFDA_URL}: {cfda_response.json()['errorMsgs']}")
+
+        return cfda_response.json()["cfdas"]
```

Comment:
When I helped with monkeypatching troubleshooting this line surfaced a problem with the code if this is an empty array (or other falsy value) which causes the API to be retried until the retries expire.

If it was my ticket I would have re-written most of this view class needs to be more robust, or at least defensively programmed, since we're dealing with an external API and cannot assume a valid response with